src/relays/TrustedRelay.sol

```diff
@@ -120,8 +132,45 @@ contract TrustedRelay {
         uint256 expiry,
         uint8 v,
         bytes32 r,
-        bytes32 s
+        bytes32 s,
+        address to,
+        bytes calldata data
     ) external {
+        uint256 startGas = gasleft();
+
+        // Withdraw the L1 DAI to the receiver
+        requestMint(wormholeGUID, signatures, maxFeePercentage, gasFee, expiry, v, r, s);
+
+        // Send the gas fee to the relayer
+        dai.transfer(msg.sender, gasFee);
+
+        // Optionally execute an external call
+        if(to != address(0)) {
+            (bool success,) = to.call(data);
+            if (!success) {
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    returndatacopy(0, 0, returndatasize())
+                    revert(0, returndatasize())
+                }
+            }
+        }
```

Comment:
What is this here for?

Reply:
That's something li.finance requested. They would like a way to compose the DAI withdrawal with a call to a custom contract (e.g. to do an AMM swap after the DAI withdrawal for instance). There are other designs to achieve that functionality (like having the relayer call a multi-call contract that batches (1) a call to the TrustedRelay and (2) a call to a custom contract) but that seemed like a simpler solution.

I think that a re-entrancy guard is effectively already performed by WormholeJoin._mint() which does an early return when being called more than once (since `wormholes[hashGUID].pending` is guaranteed to be zero after the first call).